src/tests/test_custom_commands.rs

```diff
@@ -145,3 +145,14 @@ fn override_table_eval_file() {
     let actual = nu!(cwd: ".", r#"def table [] { "hi" }; table"#);
     assert_eq!(actual.out, "hi");
 }
+
+#[test]
+fn recursion_successful() {
+    let actual = nu!(
+        cwd: ".",
+        r#"
+            def bang [] { bang }
```

Comment:
```suggestion
            def bang [] { bang }
            bang
```

I think you forgot to run `bang` in the test.